lib/git-process.ts

```diff
@@ -311,3 +332,33 @@ function ignoreClosedInputStream(process: ChildProcess) {
     }
   })
 }
+
+export interface IGitTask {
+  pid?: number
+  readonly result: Promise<IGitResult>
+  setPid(pid: number): void
+  readonly cancel: () => boolean
+}
```

Comment:
This is what will be seen by `dugite` clients, and those don't need to know about the pid, so it can be removed from here. `dugite` clients only need to worry about `result` and `cancel`, right?

Also, since this interface is public, we should include some comments documenting it:
```suggestion
/** This interface represents a git task (process). */
export interface IGitTask {
  /** Result of the git process. */
  readonly result: Promise<IGitResult>
  
  /** Allows to cancel the process if it's running. Returns true if the process was killed. */
  readonly cancel: () => boolean
}
```